### PR TITLE
fix(component): filter in list toolbar had hardcoded placeholder

### PR DESCRIPTION
**Pull request by jlame**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
The placeholder for the filter input was hardcoded 'Filter'

**What is the new behavior?**
The filter component uses the placeholder in its props if it's present otherwise it uses the default placeholder 'Filter'

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration
path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://228.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://228.talend.surge.sh)

File: packages/components/src/List/Toolbar/Filter/Filter.component.js [hunk 1]
```diff
@@ -30,13 +30,14 @@ function FilterInput(props) {
...
 		onToggle,
+		placeholder
 	} = props;
 
 	const inputProps = {
 		id,
 		name: 'search',
 		type: 'search',
-		placeholder: 'Filter',
+		placeholder: placeholder ? placeholder : 'Filter',
```
Comment (frassinier) [inline packages/components/src/List/Toolbar/Filter/Filter.component.js]: You can use `placeholder || 'Filter'` here and use [defaultProps](https://facebook.github.io/react/docs/react-component.html#defaultprops) instead

File: output/components.eslint.txt [hunk 2]
```diff
@@ -16,6 +16,15 @@ Scoping to packages that match 'react-talend-components'
   2:1  error  '@kadira/react-storybook-addon-info' should be listed in the project's dependencies, not devDependencies  import/no-extraneous-dependencies
   7:2  error  Unexpected require()                                                                                      global-require
 
+/home/travis/build/Talend/ui/packages/components/src/List/Toolbar/Filter/Filter.component.js
```
Comment (frassinier) [inline output/components.eslint.txt]: You have some lint issues to fix before merge

Comment (jsomsanith) [inline output/components.eslint.txt]: Please enable `eslint` in your IDE

File: packages/components/src/List/Toolbar/Filter/Filter.component.js (hunk 1 above)
Comment (jsomsanith) [inline packages/components/src/List/Toolbar/Filter/Filter.component.js]: 👍 
So only 
```
const inputProps = {
  		id,
  		...
		placeholder,
```

and default props

File: packages/components/src/List/Toolbar/Filter/Filter.snapshot.test.js [hunk 3]
```diff
@@ -39,4 +39,27 @@ describe('Filter', () => {
...
+			docked: false,
+			placeholder: 'find something',
+		};
+		const wrapper = renderer.create(<Filter {...defaultProps}/>).toJSON();
+
+		// then
+		expect(wrapper).toMatchSnapshot();
+	});
+
+	it('should render filter input with defaut placeholder', () => {
```
Comment (jsomsanith) [inline packages/components/src/List/Toolbar/Filter/Filter.snapshot.test.js]: Typo : `defaut ` --> `default `
Don't forget to update the snapshots after this fix

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://228.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://228.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://228.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://228.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://228.talend.surge.sh)

**Pull request merged**